extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSink.java

```diff
@@ -57,12 +62,9 @@ protected StreamResult<Object> transferParts(List<DataSource.Part> parts) {
                         .key(key)
                         .build()).uploadId();
 
-                while (true) {
-                    var bytesChunk = input.readNBytes(chunkSize);
+                var bytesChunk = input.readNBytes(chunkSize);
 
-                    if (bytesChunk.length < 1) {
-                        break;
-                    }
+                while (bytesChunk.length > 0) {
```

Comment:
to avoid double call to `readNBytes` the assignment could be done in the while clause:
```
                byte[] bytesRead;
                while ((bytesRead = input.readNBytes(chunkSize)).length > 0) {
```